### PR TITLE
Add PolyScheduler which mimics the caffe polynomial lr scheduler

### PR DESCRIPTION
**Pull request by bzcheeseman**
## Description ##
(Brief description on what this PR is about)
In reproducing a network originally defined in Caffe, I realized that MXNet had no equivalent LR scheduler. As a user of MXNet, I would like to be able to use such an LR scheduler, so I added one.

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- No tests exist currently for LR Scheduler(s), willing to add some if that is deemed useful.
- [ ] Code is well-documented: 
- Matches previous LR scheduler implementation, also willing to add some if deemed useful.
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Add `PolyScheduler` in `cpp-package/include/mxnet-cpp/lr_scheduler.h`

## Comments ##
- Matches quite closely the FactorScheduler implementation just above, but am happy to make other changes.

### CODE REVIEW

**Comment by chinakook** [comment]
I think a python version is better.

**Comment by bzcheeseman** [comment]
I mean I think something similar exists in python...the point of the PR was to add that functionality to C++ since that's what I was using and that's what I needed..?

**Comment by bzcheeseman** [comment]
I was trying to keep the naming consistent with the FactorScheduler (when I opened the PR the python one didn't exist :P) and the implementation was based on the Caffe implementation for my particular use case. I've changed it to be in line with the python version.

**Comment by bzcheeseman** [comment]
I didn't see any schedulers that had tests in the cpp package, can you point me at an example of what to test?

**Comment by stu1130** [comment]
@anirudhacharya Could you please point out the example? Thanks

**Comment by stu1130** [comment]
@mxnet-label-bot [pr-awaiting-review]

**Comment by anirudhacharya** [comment]
@bzcheeseman in this file - https://github.com/apache/incubator-mxnet/blob/master/cpp-package/tests/ci_test.sh#L27 all the examples in the examples folder are being run as part of cpp tests. you can write a test either in example or test folder and call it from the above bash script file.

**Comment by anirudhacharya** [comment]
@leleamol @nswamy for review and merge.

**Comment by vrakesh** [comment]
@leleamol  and @nswamy 
Requesting a review and merge on this issue

**Comment by leleamol** [comment]
@Roshrini  Can you please add a test or an example to exercise this code? 
Also please add the new test to ci_test.sh file.

**Comment by bzcheeseman** [comment]
I was never able to find an applicable test to base a new one off of (unless one was added in the last few days) :(

**Comment by ankkhedia** [comment]
@bzcheeseman Sorry to hear that :(

@leleamol Could you please help @bzcheeseman if you know of any applicable test/example on to how to proceed further.

**Comment by leleamol** [comment]
@bzcheeseman 
There are  examples specific to training workflow (such as alexnet, mlp_cpu, mlp_csv, resnet, lenet, etc) in the mxnet-cpp/example folder.

Using those, would you be able to come up with an example and test to exercise your code?
If you have any questions, please post them here so that community can help guide you.

Thanks.

**Comment by bzcheeseman** [comment]
@leleamol I don't really need this code anymore - I already got what I needed...I can hand this off to someone else or just close the PR?

**Comment by stu1130** [comment]
@sandeep-krishnamurthy could you close the PR?
